src/AzureIoTHub.Portal.Client/Pages/DeviceModels/CreateDeviceModelPage.razor

```diff
@@ -349,13 +501,66 @@
 
             Snackbar.Add("Device model successfully created.", Severity.Success);
 
+
+            // Go back to the list of devices
+            NavigationManager.NavigateTo("device-models");
+        }
+
+        catch (ProblemDetailsException exception)
+        {
+            Error?.ProcessProblemDetails(exception);
+        }
+        finally
+        {
+            isProcessing = false;
+        }
+    }
+    private async Task saveThingType()
+    {
+        isProcessing = true;
+
+        // Displays validation error message for each field
+        await form.Validate();
+
+
+        if (!thingTypeValidator.Validate(ThingType).IsValid)
+        {
+            Snackbar.Add("One or more validation errors occurred", Severity.Error);
+
+            isProcessing = false;
+
+            return;
+        }
+
+        try
+        {
+
+            if(this.ThingTypeTags != null)
+            {
+                if(this.ThingTypeTags.Count() > 0)
+                {
+                    foreach (var tag in this.ThingTypeTags)
+                    {
+                        this.Tags.Add(tag.Key, tag.Value);
+                    }
+                }
+            }
+
+            ThingType.Tags = this.Tags;
+            ThingType.ThingTypeSearchableAttDtos = searchableAtt;
+
+            await ThingTypeClientService.CreateThingType(ThingType);
+            Snackbar.Add("Thing Type successfully created.", Severity.Success);
+
             // Go back to the list of devices
             NavigationManager.NavigateTo("device-models");
         }
 
         catch (ProblemDetailsException exception)
         {
             Error?.ProcessProblemDetails(exception);
+            //Reinitialize Tags When an error occured
+            this.Tags = new Dictionary<string, string>();
```

Comment:
Why ?

Reply:
Because when I get the error, the page is not refreshed and I get The list of tags that is not empty and when I refill tags and put a key that already exists in this list of tags, there is another error as that key already exists. To avoid this. I just reinitialized the list so that no problems for the new values that will be entered